compiler_gym/envs/compiler_env.py

```diff
@@ -269,23 +275,14 @@ def __init__(
             for space in self.service.action_spaces
         ]
         self.observation = self._observation_view_type(
-            get_observation=lambda req: self.service(
-                self.service.stub.GetObservation, req
-            ),
+            get_observation=lambda req: self.service(self.service.stub.Step, req),
```

Comment:
BTW, `Step` sounds like taking an action, when there might be no action?
I was wondering if the right abstraction is to think of both actions and observations as the same thing - functions that get called on the compiler side. Then the service protocol is to apply a list of functions and return the results. This would allow, for example, doing [compile-at-O3, InsnCount, compile-at-somethingelse, InsnCount].
By splitting actions and observations, you can't do different observations after different actions.
But probably I'm talking guff, since I also like the idea of asking questions about code after compilation strategies have been applied.

Reply:
I agree that `Step()` isn't the right name for this service method. Happy to rename to something more clear.

I think that supporting multiple `(<action,observation>, <action, observation>...)` in a single call is overkill for now while we still have a stateful service backend that could just use multiple calls

Reply:
Aye, not a critical change :) Yep, I agree with the overkill thing.

I was thinking of a name like Apply. As in function application. Not sure if that's much better.